sumologic/resource_sumologic_hierarchy.go

```diff
@@ -239,12 +239,15 @@ func resourceToHierarchy(d *schema.ResourceData) Hierarchy {
 	rawLevel := d.Get("level").([]interface{})[0]
 	level := getLevel(rawLevel.(map[string]interface{}))
 
-	return Hierarchy{
-		Name:   d.Get("name").(string),
-		ID:     d.Id(),
-		Filter: &filter,
-		Level:  level,
+	hierarchy := Hierarchy{}
+	hierarchy.Name = d.Get("name").(string)
+	hierarchy.ID = d.Id()
+	hierarchy.Level = level
+	if len(filter.Key) > 0 {
+		hierarchy.Filter = &filter
```

Comment:
Update acceptance to test for empty filter?

Reply:
Done